cdisc_rules_engine/models/dictionaries/abstract_factory.py

```diff
@@ -0,0 +1,37 @@
+from typing import Type
+
+from cdisc_rules_engine.models.dictionaries import (
+    DictionaryTypes,
+    TermsFactoryInterface,
+)
+from cdisc_rules_engine.models.dictionaries.meddra import MedDRATermsFactory
+from cdisc_rules_engine.models.dictionaries.whodrug import WhoDrugTermsFactory
+from cdisc_rules_engine.services.data_services import BaseDataService
+from cdisc_rules_engine.services.factory_interface import FactoryInterface
+
+
+class AbstractTermsFactory(FactoryInterface):
```

Comment:
Should the MedDRATermsFactory and WhoDrugTermsFactory inherit from AbstractTermsFactory?

Reply:
Nevermind, I see this is more of a factory factory (A factory that produces factories.)